src/app/public/src/modules/code-block/code-block.component.ts

```diff
@@ -76,6 +77,13 @@ export class StacheCodeBlockComponent implements AfterViewInit {
     this.cdRef.detectChanges();
   }
 
+  public handleClipboardButton() {
+    this.clipboardButtonClicked = true;
+    setTimeout(() => {
```

Comment:
Is there a more "graceful" way to handle this? What happens if I spammed the copy button?

It would also be nice to show a "Copied!" message near the button, much like this demo:
https://clipboardjs.com/
<img width="79" alt="screen shot 2018-07-05 at 12 13 31 pm" src="https://user-images.githubusercontent.com/12497062/42335007-f531ee56-804c-11e8-98e0-7a0bba44c61c.png">

Maybe that could be accomplished with the SKY UX popover?
https://developer.blackbaud.com/skyux/components/popover

Reply:
That was the design I initially went with, but Brandon and I decided it would be more efficient to just have the icon update. This removes the need to add translation text and also gives the button a smaller footprint on the page, since there's no popover when you click. I'd be happy to discuss this further if you have additional thoughts.